test/unit/controllers/UploadWrittenAuthorisationControllerSpec.scala

```diff
@@ -136,6 +137,20 @@ class UploadWrittenAuthorisationControllerSpec extends ControllerSpecBase with M
       status(result) mustBe BAD_REQUEST
     }
 
+    "return a Bad Request when invalid file type is submitted" in {
+
+      val file = TemporaryFile("example-file.mp3")
+      val filePart = FilePart[TemporaryFile](key = "letter-of-authority", "example-file.mp3", contentType = Some("audio/mpeg"), ref = file)
+      val form = MultipartFormData[TemporaryFile](dataParts = Map(), files = Seq(filePart), badParts = Seq.empty)
+      val postRequest = fakeRequest.withBody(form)
+
+      given(fileService.validate(any[MultipartFormData.FilePart[TemporaryFile]])).willReturn(Left("bad file"))
+
+      val result = controller().onSubmit(NormalMode)(postRequest)
+
+      status(result) mustBe BAD_REQUEST
```

Comment:
how about the body of the response?

Reply:
Looks like he is below @gokyo

Reply:
I added that assertion after Stefano's comment.